lib/std/fs/test.zig

```diff
@@ -687,6 +687,49 @@ test "makePath in a directory that no longer exists" {
     try testing.expectError(error.FileNotFound, tmp.dir.makePath("sub-path"));
 }
 
+fn testFilenameLimits(iterable_dir: IterableDir, maxed_filename: []const u8) !void {
+    // setup, create a dir and a nested file both with maxed filenames, and walk the dir
+    {
+        var maxed_dir = try iterable_dir.dir.makeOpenPath(maxed_filename, .{});
+        defer maxed_dir.close();
+
+        try maxed_dir.writeFile(maxed_filename, "");
+
+        var walker = try iterable_dir.walk(testing.allocator);
+        defer walker.deinit();
+
+        var count: usize = 0;
+        while (try walker.next()) |entry| {
+            try testing.expectEqualStrings(maxed_filename, entry.basename);
+            count += 1;
+        }
+        try testing.expectEqual(@as(usize, 2), count);
+    }
+
+    // ensure that we can delete the tree
+    try iterable_dir.dir.deleteTree(maxed_filename);
+}
+
+test "filename limits" {
+    var tmp = tmpIterableDir(.{});
+    defer tmp.cleanup();
+
+    if (builtin.os.tag == .windows) {
+        // € is the character with the largest codepoint that is encoded as a single u16 in UTF-16,
+        // so Windows allows for NAME_MAX of them
+        const maxed_windows_filename = ("€".*) ** std.os.windows.NAME_MAX;
+        try testFilenameLimits(tmp.iterable_dir, &maxed_windows_filename);
+    } else if (builtin.os.tag == .wasi) {
+        // On WASI, the maxed filename depends on the host OS, so in order for this test to
```

Comment:
I'm not quite sure what you mean. Is there a way to detect what the host OS is when running via WASI? If not, then we can't know what the 'real' `MAX_NAME` is, so we need to test with a file name component length that will work for any host.

The issue this is trying to get around is that the non-Windows branch of the test uses `** MAX_NAME_BYTES` which for WASI would give the Windows max of 765, which would be too large when WASI is run on a non-Windows host OS (i.e. the makeDir and writeFile calls would fail).

Reply:
Ah, my main concern was that the code looked different, so its not obvious which OS has the biggest filename.

As for the other concerns to minimize necessary size in case Windows is not a WASI target: 
There could be a function, that takes a list of user-provided targets and returns the maximum necessary size. Alternatively, the creation of such a function could be described and the user provides the sizes as comptime (or runtime) argument.

However, I think this should be a separate PR.